modules/markup/renderer.go

```diff
@@ -239,8 +265,11 @@ func (err ErrUnsupportedRenderExtension) Error() string {
 }
 
 func renderFile(ctx *RenderContext, input io.Reader, output io.Writer) error {
-	extension := strings.ToLower(filepath.Ext(ctx.Filename))
+	extension := strings.ToLower(filepath.Ext(ctx.RelativePath))
 	if renderer, ok := extRenderers[extension]; ok {
+		if renderer.DisplayInIFrame() && ctx.AllowIFrame {
+			return renderIFrame(ctx, renderer, input, output)
+		}
```

Comment:
One more question , if `renderFile` is called somewhere and it matches an external render without `AllowIFrame=true`, will the content be rendered directly in current page? If yes, is it allowed or should it be forbidden?

Reply:
In fact, `AllowIFrame=false, DisplayInIFrame=true` is used by the link in the iframe.

Reply:
I mean "(a normal page, not the /render/*)", is it guaranteed that no other code would call `renderFile` with "AllowIFrame=false, DisplayInIFrame=true" ? If it happens, the content be rendered directly in normal page.